app/models/chargeable_field.rb

```diff
@@ -70,6 +70,13 @@ def metric_key(sub_metric = nil)
     "#{rate_name}_#{sub_metric ? sub_metric + '_' : ''}metric" # metric value (e.g. Storage [Used|Allocated|Fixed])
   end
 
+  # Fixed metric has _1 or _2 in name but column
+  # fixed_compute_metric is used in report and calculations
+  # TODO: remove and unify with metric_key
+  def metric_column_key
+    fixed? ? metric_key.gsub(/\_1|\_2/, '') : metric_key
```

Comment:
I'm not sure what this is doing

Reply:
It is hack,

we have chargeback_rate_details and each detail is representing metric column na cost column.

For example:

CPU Allocated Cost has metric column: `cpu_allocated_metric` and cost column `cpu_allocated_cost`

so when we are checking if `cpu_allocated_cost` is allowed to report, we are asking here https://github.com/ManageIQ/manageiq/pull/17414/files/213289cd034ea8054bbb9a30c80d229cd776c049#diff-7c998acd3d11c70ed830e216754f4b6bR42 
if string `cpu_allocated_cost` (`from selected fields`) is related to `metric_key` which is based from `ChargeableField#group and Chargeback#source`.

But we have to add exception for `fixed rate detail` because there are two columns
`fixed_compute_1_cost` and `fixed_compute_1_cost` but related metric field is only one: `fixed_compute_metric` and then method ChargeableField#metric_key is not enough  (it would produce `fixed_compute_1_metric `) but we need to base check in `fixed_compute_metric ` (some for storage)

so input of this method is `fixed_compute_1_metric` and it should remove `_1` (etc.)